datalad/support/exceptions.py

```diff
@@ -60,6 +60,17 @@ def to_str(self, include_output=True):
         if self.stderr:
             to_str += " [err: '{}']".format(ensure_unicode(self.stderr).strip())
         if self.kwargs:
+            if 'stdout_json' in self.kwargs:
+                from datalad.utils import unique
+                json_errors = unique([
+                    '{}{}'.format(
+                        m['note'] if m.get('note') else m.get('error-messages'),
+                        m['error-messages'] if m.get('error-messages') else '')
```

Comment:
In the case that 'note' is false and 'error-messages' is not, wouldn't the string formatted as '{error-messages}{error-messages}'?  Admittedly, I can't come up with a git-annex call that has an error-messages field but no note.

Aside from being confused by that, this PR looks like a clear improvement to me.

Reply:
Yeah, that is probably an unlikely event, but still needs fixing IMHO. Will do. Thx!